Scrabble.java

```diff
@@ -85,16 +111,34 @@ public static void playHand(String hand) {
 			// non-whitespace characters. Whitespace is either space characters, or  
 			// end-of-line characters.
 			String input = in.readString();
-			//// Replace the following break statement with code
-			//// that completes the hand playing loop
-			break;
-		}
-		if (hand.length() == 0) {
-	        System.out.println("Ran out of letters. Total score: " + score + " points");
-		} else {
-			System.out.println("End of hand. Total score: " + score + " points");
+			
+			if (input.equals(".")){
+				break;	
+
+			}if (MyString.subsetOf(input, hand)){
+				if (isWordInDictionary(input)){
+					hand = MyString.remove(hand, input);
+					int wordScore = wordScore(input);
+					score += wordScore;
+					System.out.println(input + " earned " + wordScore + " points. Score: " + score + " points\n");
+
+				
+				}else{
+					System.out.println("No such word in the dictionary. Try again.");
+				
+			}

```

Comment:
Handle the not cases then the true cases 
for better practice and reduce scopes